docs/conf.py

```diff
@@ -303,14 +303,53 @@
 # a conflict with the sphinx-docsearch extension for Algolia search
 
 nbsphinx_requirejs_path = ""
+
+# For notebook downloads (23.5 onwards), we get the version from the environment
+# variable READTHEDOCS_VERSION and set it accordingly.
+
+# If the version is set to "latest", then we are on the develop branch, and we
+# point to the notebook in the develop blob
+# If we are on "stable", we point to the notebook in the relevant release tree
+# for the PyBaMM version
+# On a PR build, we use READTHEDOCS_GIT_COMMIT_HASH which will always point to changes
+# made to a notebook, if any.
+# On local builds, the version is not set, so we use "latest".
+
+if (os.environ.get("READTHEDOCS_VERSION") == "latest") or (os.environ.get("READTHEDOCS_VERSION") is None): # noqa: E501
+    notebooks_version = "develop"
+    github_download_url = (
+        "https://github.com/pybamm-team/PyBaMM/blob/" + notebooks_version
```

Comment:
1. f-strings should be preferred for string concatenation as they are more efficient
2. It would be better to just refactor out the github URL to avoid repetition
```suggestion
    append_to_url = f"blob/{notebooks_version}"
```